src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeGenerator.java

```diff
@@ -990,11 +943,18 @@ private boolean inBlockArgs(Op.Result res) {
         return false;
     }
 
+    private static boolean moreThanOneUse(Op.Result res) {
+        Set<Op.Result> uses = res.uses();
+        return uses.size() > 1
+            || uses.size() == 1 && uses.iterator().next().op().operands().stream().filter(o -> o == res).count() > 1
+            || inBlockArgs(res);
```

Comment:
Uses will return a use if the value used as an operand or as a successor argument, but as you found out it will not differentiate between single and multiple uses of the same value for the same operation, which in theory could be across operands or successor arguments.

Reply:
Will fix it, thanks.